connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -67,7 +67,7 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
       case proto.Relation.RelTypeCase.AGGREGATE => transformAggregate(rel.getAggregate)
       case proto.Relation.RelTypeCase.SQL => transformSql(rel.getSql)
       case proto.Relation.RelTypeCase.LOCAL_RELATION =>
-        transformLocalRelation(rel.getLocalRelation)
+        transformLocalRelation(rel.getLocalRelation, common)
```

Comment:
what is `common`? Every logical plan has an option alias?

Reply:
This is legacy design that I believe it thinks only relations have the optional alias. 

Every logical plan could have an optional alias, in that case I prefer to move that alias out of the `common` to have its own message. This is because by that we can differentiate
```
.xx()
.xx().as("") // probably invalid but user can write down such API
.xx().as("alias_1")
```  

I can also change this in this PR if you think this is a right time.

Reply:
I sent a PR for this topic (to avoid complicate current refactoring PR too much): https://github.com/apache/spark/pull/38415